### ISSUE TITLE
[java] False positive violation of SwitchStmtsShouldHaveDefault when using enhanced switch

### ISSUE THREAD

**Issue by julius-d** [pmd/pmd#2997]
**Affects PMD Version:**  6.30.0

**Description:**

Given this code:
```
public class Example {

  enum Type {
    CAT,
    DOG,
    COW
  }

  public void foo(Type type) {
    switch (type) {
      case CAT -> System.out.println("Cat");
      case DOG -> System.out.println("Dog");
      case COW -> System.out.println("Cow");
      default -> System.out.println("Don't know");
    }
  }
}

```
I get a unexpected violation

```
[INFO] PMD Failure: com.example.Example:12 Rule:SwitchStmtsShouldHaveDefault Priority:3 Switch statements should have a default label.
```

I also get the violation reported from pmd when removing the default case.

I run pmd via maven

Versions:
maven-pmd-plugin:3.14.0
PMD version: 6.30.0
java version: openjdk version "15.0.1" 2020-10-20

**Comment by oowekyala** [comment]
This is a duplicate of #2806. @adangel Maybe we shouldn't close tickets scheduled for pmd 7 before they're fixed on master?

**Comment by oowekyala** [comment]
Will keep open as #2806
